TM1py/Services/CellService.py

```diff
@@ -991,6 +1010,145 @@ def write_through_unbound_process(self, cube_name: str, cellset_as_dict: Dict, i
+                           delete_blob=True, dimensions: str = None, **kwargs):
+        """
+        Writes data back to TM1 via an unbound TI process having an uploaded CSV as data source
+        :param cube_name: str
+        :param cellset_as_dict:
+        :param increment: increment or update cell values
+        :param sandbox_name: str
+        :param skip_non_updateable skip cells that are not updateable (e.g. rule derived or consolidated)
+        :param delete_blob: choose False for debugging purposes
+        :param dimensions: optional. Dimension names in their natural order. Will speed up the execution!
+        :param kwargs:
+        :return: Success: bool, Messages: list, ChangeSet: None
+        """
+
+        process_service = ProcessService(self._rest)
+        cube_service = self.get_cube_service()
+        file_service = FileService(self._rest)
+
+        # Generate hash based on tm1-session-id and local-thread to guarantee unique name
+        unique_string = f"{self._rest.session_id}{threading.get_ident()}"
+        unique_hash = hashlib.sha256(unique_string.encode('utf-8')).hexdigest()[:12]
+
+        # Transform cells to format that's consumable for TI
+        csv_content = StringIO()
+        csv_writer = csv.writer(
+            csv_content,
+            delimiter=",",
+            quoting=csv.QUOTE_ALL)
+        csv_writer.writerows(
+            list(elements) + [value.replace('\r', '').replace('\n', '') if isinstance(value, str) else value]
+            for elements, value
+            in cellset_as_dict.items())
+
+        file_name = f'{unique_hash}.csv'
+        file_service.create(
+            file_name=file_name,
+            file_content=csv_content.getvalue().encode('utf-8'),
+            **kwargs)
+
+        try:
+            # Create and execute unbound TI process to load blob file to cube
+            process_name = f"}}tm1py.{unique_hash}"
+            process = self._create_write_process(
+                cube_name=cube_name,
+                process_name=process_name,
+                blob_filename=file_name,
+                dimensions=dimensions or cube_service.get_dimension_names(cube_name),
+                increment=increment,
+                skip_non_updateable=skip_non_updateable,
+                sandbox_name=sandbox_name,
+                **kwargs)
+
+            success, status, log_file = process_service.execute_process_with_return(process=process, **kwargs)
+            if not success:
+                if status in ['HasMinorErrors']:
+                    raise TM1pyWritePartialFailureException([status], [log_file], 1)
+                else:
+                    raise TM1pyWriteFailureException([status], [log_file])
+
+        finally:
+            if delete_blob:
+                file_service.delete(file_name=file_name)
+
+    def _create_write_process(self, cube_name: str, process_name: str, blob_filename: str, dimensions: List[str],
```

Comment:
If this is only used for blobs and it doesn't actually create a process on the server, should we be more specific on the naming? Something like `_build_write_process_using_blob` or `_build_write_process_from_blob`?